bin/pycbc_compress_bank

```diff
@@ -136,17 +142,31 @@ for ii in range(imin, imax):
     bank.delta_f = df
     bank.N = N
     bank.filter_length = N/2 + 1
-    # We'll budge the waveform starting frequency down by 1df to ensure 
-    # the first point is correct
-    bank.f_lower = fmin-df
+    bank.f_lower = fmin
     bank.kmin = int(bank.f_lower / df)
     # scratch space
     decomp_scratch = FrequencySeries(numpy.zeros(N, dtype=dtype), delta_f=df)
     # generate the waveform
     htilde = bank[ii-imin]
+    if numpy.abs(htilde[bank.kmin]) == 0:
+        raise ValueError("""The amplitude of the waveform at the
+                         low_frequency_cutoff is zero. A non-zero value
+                         is required.""")
     tmplt = bank.table[ii-imin]
     kmax = htilde.end_idx
-    
+    if args.psd_from_file :
```

Comment:
Now, down here, you can just use [pycbc.psd.from_cli](https://github.com/ligo-cbc/pycbc/blob/master/pycbc/psd/__init__.py#L28). Note that that function will return None if none of the psd options were set, which is what you want. It will also take care of including a dynamic range factor, which is important when using single precision.

One other thing: it would be faster if you cache the PSDs, instead of regenerating them for every template. You only need a different psd for each `N` that is used. So, create a dictionary to store the psds outside of the loop. Then, when you go to get the PSD:
```
try:
    psd = psd_dict[N]
except KeyError:
   psd = pycbc.psd.from_cli(...)
   psd_dict[N] = psd
```